File: Front-End/src/services/variant.service.ts
import axiosClient from '@/configurations/axios.config';
import type { 
  CreateVariantRequest, 
  VariantResponse, 
  VariantListResponse,
  VariantValueResponse,
  VariantListResponseForCreateProduct
} from '@/types/variant.type';

export const variantService = {
  getVariants: async (page: number = 1, size: number = 10, search: string = "") => {
    const response = await axiosClient.get<VariantListResponse>(`/variants?page=${page}&size=${size}&variantName=${search}`);
    return response.data;
  },
  getVariantsForCreateProduct: async (categoryId: number) => {
    const response = await axiosClient.get<VariantListResponseForCreateProduct>(`/variants/category/${categoryId}`);
    return response.data;
  },

  getVariantById: async (id: number) => {
    const response = await axiosClient.get<VariantResponse>(`/variants/${id}`);
    return response.data;
  },

  createVariant: async (request: CreateVariantRequest) => {
    const response = await axiosClient.post<VariantResponse>('/variants', request);
    return response.data;
  },

  updateVariant: async (id: number, request: CreateVariantRequest) => {
    const response = await axiosClient.put<VariantResponse>(`/variants/${id}`, request);
    return response.data;
  },

  changeStatusVariant: async (id: number) => {
    await axiosClient.put(`/variants/change-status/${id}`);
  },

  getVariantValuesByVariantId: async (id: number) => {
    const response = await axiosClient.get<VariantValueResponse>(`/variant-values/variant/${id}`);
    return response.data;
  }
};

File: Back-End/src/main/java/iuh/fit/ecommerce/services/VariantService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.dtos.request.variant.VariantAddRequest;
import iuh.fit.ecommerce.dtos.response.base.ResponseWithPagination;
import iuh.fit.ecommerce.dtos.response.variant.VariantResponse;

import java.util.List;

public interface VariantService {
    VariantResponse createVariant(VariantAddRequest request);

    ResponseWithPagination<List<VariantResponse>> getVariants(int page, int size, String variantName);

    VariantResponse getVariantById(Long id);

    VariantResponse updateVariant(Long id, VariantAddRequest request);

    void changeStatusVariant(Long id);

    List<VariantResponse> getVariantsByCategory(Long id);
}
